File: transpose.R
transpose <- function(matrix_values){
    COL_LENGTH <- ncol(matrix_values)
    ROW_LENGTH <- nrow(matrix_values)


    transpose_arr  <- matrix(data = 0, nrow = COL_LENGTH, ncol = ROW_LENGTH)

    for(row in 1:ROW_LENGTH){
        for(col in 1:COL_LENGTH){
            transpose_arr[col,row] <- matrix_values[row, col]
        }
    }
    return(transpose_arr)
}

matrix_values  <- matrix(data = 1:20, nrow = 10, ncol = 2)
transpose_matrix <- transpose(matrix_values)
print(transpose_matrix)

File: transpose.js
function transpose(arr){
    const COL_LENGTH = arr[0].length;
    const ROW_LENGTH = arr.length;
    let tranpose_arr = [];

    let temp_arr
    for(let col = 0; col < COL_LENGTH; col++){
        temp_arr = [];
        for(let row = 0; row < ROW_LENGTH; row++){
            temp_arr.push(arr[row][col])
        }
        tranpose_arr.push(temp_arr);
    }
    return tranpose_arr;
}


const arr = [
    [1, 2, 3, 7, 9],
    [4, 5, 6, 8, 10],
    [11,12,13,14,15],
];

console.table(arr);
const tranpose_arr = transpose(arr);

console.table(tranpose_arr);